### PR TITLE
Fixed formatting for nested markup extensions

### PR DESCRIPTION
**Pull request by FrecherxDachs**
- MarkupExtensionFormatter is now invoked recursively if attributes
  contain nested markup extensions
- elements with only one attribute should now respect markup
  extension formatting

### CODE REVIEW

File: XamlStyler.Core/MarkupExtensions/Formatter/MarkupExtensionFormatterBase.cs [hunk 1]
```diff
@@ -80,7 +87,7 @@ private IEnumerable<string> Format(Value value)
 
             if (type == typeof(MarkupExtension))
             {
-                return this.Format((MarkupExtension)value);
+                return this.markupExtensionFormatter.Format((MarkupExtension)value);
```
Comment (FrecherxDachs) [inline XamlStyler.Core/MarkupExtensions/Formatter/MarkupExtensionFormatterBase.cs]: recursively call markup extension formatting

File: XamlStyler.Core/DocumentProcessors/ElementDocumentProcessor.cs [hunk 2]
```diff
@@ -183,8 +183,8 @@ public void Process(XmlReader xmlReader, StringBuilder output, ElementProcessCon
                 }
             }
 
-            // No need to break attributes.
-            if (noLineBreakInAttributes)
+            // No need to break attributes if there are no markup extensions formatting.
+            if (noLineBreakInAttributes && (!this.options.FormatMarkupExtension || list.TrueForAll(info => !info.IsMarkupExtension)))
```
Comment (FrecherxDachs) [inline XamlStyler.Core/DocumentProcessors/ElementDocumentProcessor.cs]: markup extension formatting should also be applied on single attribute elements

**Comment by grochocki** [comment]
Thank you for your contribution! Unfortunately, there are 16 failing unit tests with this PR. Please take a look at the failing tests, make any appropriate changes, and update the PR when all tests are passing.

**Comment by grochocki** [comment]
Fix implemented in #72

**Pull request closed without merging**